prow/config/jobs_test.go

```diff
@@ -76,8 +79,8 @@ func TestPresubmits(t *testing.T) {
 			}
 			// Check that the merge bot will run AlwaysRun jobs, otherwise it
 			// will attempt to rerun forever.
-			if job.AlwaysRun && !job.re.MatchString(testThis) {
-				t.Errorf("AlwaysRun job %s: \"%s\" does not match regex \"%v\".", job.Name, testThis, job.Trigger)
+			if job.AlwaysRun && !job.re.MatchString(newTestThis) && !job.re.MatchString(oldTestThis) {
```

Comment:
To clarify, there will be two conditions to check:
```
if job.AlwaysRun && !job.re.MatchString(newTestThis) && !job.re.MatchString(oldTestThis)
if !job.AlwaysRun && (job.re.MatchString(newTestThis) || job.re.MatchString(oldTestThis))
```

First check ensures `AlwaysRun` jobs match _at least_ one of the commands.
Second check ensures `Non-AlwaysRun` jobs does not match _any_ of the commands.